cli/src/commands/grpc-service/commands/generate.ts

```diff
@@ -68,27 +117,114 @@ async function generateCommandAction(name: string, options: CLIOptions) {
       program.error(`Input file ${options.input} does not exist`);
     }
 
+    // Validate operations directory if provided
+    if (options.withOperations) {
+      const operationsPath = resolve(options.withOperations);
+      if (!(await exists(operationsPath))) {
+        program.error(`Operations directory ${options.withOperations} does not exist`);
+      }
+      if (!(await lstat(operationsPath)).isDirectory()) {
+        program.error(`Path ${options.withOperations} is not a directory`);
+      }
+    }
+
+    // Validate and warn about query-idempotency usage
+    if (options.queryIdempotency) {
+      if (!options.withOperations) {
+        spinner.warn('--query-idempotency flag is ignored when not using --with-operations');
+      }
+
+      const validLevels = ['NO_SIDE_EFFECTS', 'DEFAULT'];
+      const level = options.queryIdempotency.toUpperCase();
+      if (!validLevels.includes(level)) {
+        program.error(
+          `Invalid --query-idempotency value: ${options.queryIdempotency}. Valid values are: ${validLevels.join(', ')}`,
+        );
+      }
+    }
+
+    // Parse custom scalar mappings if provided
+    let customScalarMappings: Record<string, string> | undefined;
+    if (options.customScalarMapping) {
+      try {
+        customScalarMappings = await parseCustomScalarMapping(options.customScalarMapping);
+      } catch (error) {
+        program.error(
+          `Failed to parse custom scalar mapping: ${error instanceof Error ? error.message : String(error)}`,
+        );
+      }
+    }
+
+    // Parse maxDepth if provided
+    let maxDepth: number | undefined;
+    if (options.maxDepth) {
+      const parsed = Number.parseInt(options.maxDepth, 10);
+      if (Number.isNaN(parsed) || parsed < 1) {
+        program.error(`Invalid --max-depth value: ${options.maxDepth}. Must be a positive integer.`);
+      }
+      maxDepth = parsed;
+    }
+
+    // Validate prefix-operation-type usage
+    if (options.prefixOperationType && !options.withOperations) {
+      spinner.warn('--prefix-operation-type flag is ignored when not using --with-operations');
+    }
+
     const result = await generateProtoAndMapping({
       outdir: options.output,
       schemaFile: inputFile,
       name,
       spinner,
       packageName: options.packageName,
       goPackage: options.goPackage,
+      javaPackage: options.javaPackage,
+      javaOuterClassname: options.javaOuterClassname,
+      javaMultipleFiles: options.javaMultipleFiles,
+      csharpNamespace: options.csharpNamespace,
+      rubyPackage: options.rubyPackage,
+      phpNamespace: options.phpNamespace,
+      phpMetadataNamespace: options.phpMetadataNamespace,
+      objcClassPrefix: options.objcClassPrefix,
+      swiftPrefix: options.swiftPrefix,
```

Comment:
We should make this an object or class and pass this as a single element instead of having an infinite amount of parameters